lib/rucio/common/hash_algorithms/__init__.py

```diff
@@ -0,0 +1,69 @@
+from abc import ABC, abstractmethod
+from importlib import import_module
+from pathlib import Path
+from pkgutil import iter_modules
+
+
+class RucioHashAlgorithm(ABC):
+    @staticmethod
+    @abstractmethod
+    def compute_on_file(file):
+        raise NotImplementedError("Abstract method called")
+
+
+#
+# Moved from rucio.common.utils. Could be refactored a bit but for the sake
+# of backwards compatibility, leave them be for now.
+#
+
+GLOBALLY_SUPPORTED_CHECKSUMS = ["adler32", "md5"]
+PREFERRED_CHECKSUM = GLOBALLY_SUPPORTED_CHECKSUMS[0]
+CHECKSUM_ALGO_DICT = {}
+
+
+def is_checksum_valid(checksum_name):
+    """
+    A simple function to check wether a checksum algorithm is supported.
+    Relies on GLOBALLY_SUPPORTED_CHECKSUMS to allow for expandability.
+
+    :param checksum_name: The name of the checksum to be verified.
+    :returns: True if checksum_name is in GLOBALLY_SUPPORTED_CHECKSUMS list, False otherwise.
+    """
+
+    return checksum_name in GLOBALLY_SUPPORTED_CHECKSUMS
+
+
+def set_preferred_checksum(checksum_name):
+    """
+    A simple function to check wether a checksum algorithm is supported.
+    Relies on GLOBALLY_SUPPORTED_CHECKSUMS to allow for expandability.
+
+    :param checksum_name: The name of the checksum to be verified.
+    :returns: True if checksum_name is in GLOBALLY_SUPPORTED_CHECKSUMS list, False otherwise.
+    """
+    if is_checksum_valid(checksum_name):
+        global PREFERRED_CHECKSUM
+        PREFERRED_CHECKSUM = checksum_name
+
+
+#
+# End of backwards-compatibility section
+#
+
+
+package_dir = Path(__file__).resolve().parent
```

Comment:
Just to confirm: Here you want to load modules which are part of this directory?

I think it might be better to do this similar to policy packages and make these hash algorithms as part of the policy package. See https://github.com/rucio/rucio/blob/master/lib/rucio/common/schema/__init__.py for example.

Reply:
> Just to confirm: Here you want to load modules which are part of this directory?

Yes, that's the idea.